Add an Admin page that allows deletion of comments

Leoism opened the pull request:
The Admin page displays all comments that Datastore is holding and it allows to selectively delete comments that the user wishes to delete. All comments selected by the user will be deleted on the Datastore.

Commit:
Add an Admin page that allows deletion of comments
The Admin page displays all comments that Datastore is holding and it allows to selectively delete comments that the user wishes to delete.

Commit:
Add an Admin page that allows deletion of comments
The Admin page displays all comments that Datastore is holding and it allows to selectively delete comments that the user wishes to delete.

Commit:
Update class description

Commit:
Update class description

andrewnguyen commented on leoism/step/portfolio/src/main/java/com/google/sps/servlets/AdminServlet.java:
```diff
@@ -0,0 +1,98 @@
+      for (CommentInformation comment : commentsInformation) {
+        // At the moment, the timestamp is closest unique element for each
+        // comment.
+        if (comment.getTimestamp() == dataStoreTimestamp) {
```
can you define an id for comments?

Leoism replied:
Done

Commit:
Store Key in Comment Entity

Commit:
Store Key in Comment Entity

Commit:
Store datastore key within the Comment Entity
Storing a key allows having a more accurate way of identifying Commments and allows a more efficient way of deleting comments

Commit:
Store datastore key within the Comment Entity
Storing a key allows having a more accurate way of identifying Commments and allows a more efficient way of deleting comments

andrewnguyen commented on leoism/step/portfolio/src/main/java/com/google/sps/servlets/AdminServlet.java:
```diff
@@ -0,0 +1,88 @@
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    Gson gson = new Gson();
```
do you also need to auth the user in doPost also?

Leoism replied:
When the user accesses the '/admin' page they are required to sign in. The only post requests done are within the AdminPage.jsp page, which is only accessible once a user is authenticated. Do you recommend requiring an authentication here?

andrewnguyen replied:
yes- someone who knows the url could just post to this endpoint couldn't they?

Leoism replied:
True! done

andrewnguyen commented on leoism/step/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -75,27 +77,44 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
+    // To retrieve a proper key, the Datastore must first perform its 'put'
+    // batch actions to generate a key. Once the key is generated store it to
+    // allow identifying messages.
+    commentEntity.setProperty("key",
```
Do you really need to do this? Seems like you can create a key first and set it as the key of the commentEntity using Entity builder https://cloud.google.com/datastore/docs/concepts/entities#creating_an_entity

Leoism replied:
Using an Entity builder is not possible because that belongs to the Google Cloud Datastore API and I am using the AppEngine Datastore API. Importing the Google Cloud Datastore API would cause confusion since there are classes with the same names between the two APIs.

andrewnguyen replied:
That's fine then. However reading the docs it seems like there's already a key assigned to the commentEntity then. Why are you setting another property called "key" on it?

Leoism replied:
That is true, an Entity already has a property and method to retrieve a key. At first, I was not sure that iterating an iterable was possible in JSTL. Documentation said it could only iterate iterators and not iterables. Luckily, PreparedQuery has a method to convert it into an iterator.

I was able to implement it and now key is no longer necessary to store :)

andrewnguyen commented on leoism/step/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -51,9 +54,8 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
-
-      CommentInformation commentInformation =
-          new CommentInformation(comment, name, timestamp, sentimentScore);
+      String key = (String) entity.getProperty("key");
```
can this just be  ```Key key = entity.getKey()```?

dchin2 committed:
Add linters. (#50)

* Add linters.

Commit:
Remove key property from Entity
The Entity has a method to retrieve its key. To use it, the AdminPage JSP has to calculate it whenever the page is called by calling the keyToString method from KeyFactory

Commit:
Remove key property from Entity
The Entity has a method to retrieve its key. To use it, the AdminPage JSP has to calculate it whenever the page is called by calling the keyToString method from KeyFactory

andrewnguyen reviewed (commented):
Looks good. Just fix the lint errors that make sense and I'll approve

Commit:
Format Java files

Commit:
Address merge conflicts
